jscomp/core/js_exp_make.ml

```diff
@@ -202,7 +202,18 @@ let make_block ?comment tag tag_info es mutable_flag : t =
       expression_desc = Object property_map;
       comment 
     }
-  | _ -> make_block ?comment tag tag_info es mutable_flag 
+  | Blk_module Some(des) ->
+    let des = Array.of_list des in
+    let property_map = List.mapi (fun i e -> (des.(i), e)) es in
+    {
+      expression_desc = Object property_map;
```

Comment:
The allocation of des can be avoided, there is Ext_list.map2

Reply:
Sometimes des is a longer list, not sure why.

Reply:
There was an issue when composing coercions. Fixed now.